lightning/src/ln/msgs.rs

```diff
@@ -1489,56 +1478,51 @@ impl Readable for OnionHopData {
 	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
 		let b: BigSize = Readable::read(r)?;
 		const LEGACY_ONION_HOP_FLAG: u64 = 0;
-		let (format, amt, cltv_value) = if b.0 != LEGACY_ONION_HOP_FLAG {
-			let mut rd = FixedLengthReader::new(r, b.0);
-			let mut amt = HighZeroBytesDroppedBigSize(0u64);
-			let mut cltv_value = HighZeroBytesDroppedBigSize(0u32);
-			let mut short_id: Option<u64> = None;
-			let mut payment_data: Option<FinalOnionHopData> = None;
-			let mut keysend_preimage: Option<PaymentPreimage> = None;
-			decode_tlv_stream!(&mut rd, {
-				(2, amt, required),
-				(4, cltv_value, required),
-				(6, short_id, option),
-				(8, payment_data, option),
-				// See https://github.com/lightning/blips/blob/master/blip-0003.md
-				(5482373484, keysend_preimage, option)
-			});
-			rd.eat_remaining().map_err(|_| DecodeError::ShortRead)?;
-			let format = if let Some(short_channel_id) = short_id {
-				if payment_data.is_some() { return Err(DecodeError::InvalidValue); }
-				OnionHopDataFormat::NonFinalNode {
-					short_channel_id,
-				}
-			} else {
-				if let &Some(ref data) = &payment_data {
-					if data.total_msat > MAX_VALUE_MSAT {
-						return Err(DecodeError::InvalidValue);
-					}
-				}
-				OnionHopDataFormat::FinalNode {
-					payment_data,
-					keysend_preimage,
-				}
-			};
-			(format, amt.0, cltv_value.0)
+		if b.0 == LEGACY_ONION_HOP_FLAG {
```

Comment:
How would should we handle this?

Reply:
Nice thanks! [3778824](https://github.com/lightningdevkit/rust-lightning/pull/1413/commits/377882446afb3ed73339710336d30d2317c81dcf)